src/main/java/com/whiteclarkegroup/liquibaselinter/report/TextReporter.java

```diff
@@ -0,0 +1,137 @@
+package com.whiteclarkegroup.liquibaselinter.report;
```

Comment:
This serves as a base reporter for both the `ConsoleReporter` and the `MarkdownReporter`. If I remove this class, there will be duplication required between `ConsoleReporter` and `MarkdownReporter`. 

If I make it abstract, I can move a number of the different `print...` methods into `ConsoleReporter`, but not all of them. I thought that the general layout that it prescribes could be valuable to custom reporters too that want to be just like the `ConsoleReporter`, but just a little different. 

Eliminating the `TextReporter` could potentially significantly increase what custom reporters are required to implement.

Reply:
Have I convinced you it is worth leaving as I designed?

Reply:
> This serves as a base reporter for both the ConsoleReporter and the MarkdownReporter. If I remove this class, there will be duplication required between ConsoleReporter and MarkdownReporter.

I probably don't mind the duplication but if it's a real necessity for reporters then we could consider a utils class.

Reply:
I guess I'm not seeing what advantage a utils class holds over a base class in this case. If you really want to eliminate the option to produce a plain text report, then fine. I'm not planning to use it either. But there is a real distinction between the `AbstractReporter`, which doesn't assume anything about where the report is being output (e.g. it could be to a database table) and the `TextReporter`, which does assume something about the report going to a file (via `PrintWriter`) and sets up a common structure of the report organized by change log, then change set, then type (`ERROR`, `IGNORED`, ...). 

It is that common structure that will be all duplication between the two reports if we don't keep the `TextReporter` base class. Even the Utils class isn't going to let me keep that structure without passing in a whole bunch of different Lambdas into the Utils class. Lambdas are great and all, and this project uses them plenty, but that seems like a strange solution just to avoid making a base class.

I'm fine with renaming the `TextReporter` to something else. I'm fine making it abstract, in which case, there are some methods that can be moved/removed. I just don't see the benefit in eliminating it altogether.

Reply:
Thanks for elaborating. I see your point re an abstract class you can extend to build a text-based reporter. I'm going to dig into this a little more today but it seems like a fair direction.